### DISCUSSION TITLE
Micro-time resolution (upsampling before convolution)

### DISCUSSION THREAD

**Discussion by Masharipov** [cvnlab/GLMsingle#105] [Q&A] [answered]
Hi! I wonder what micro-time resolution you use in the GLMsingle toolbox? By default, SPM uses dt = TR/16, FSL uses dt=0.05s, and AFNI uses 0.01s. I was unable to find these values in your MATLAB code. I read that upsampling should be *on* by default as part of HRF convolution.

**Comment by kendrickkay** [comment, marked as answer]
GLMsingle is fairly agnostic.  You can prepare your design matrix and fMRI data at whatever time resolution you want.  It is, I suppose, implied that you do not want to go to the micro-scale, since the expensive operations/computations are carried at whatever resolution you supply.  In terms of the low pass properties of the HRF, a rough guess is that going more fine than 1 s will likely not buy you much benefit.    We do recommend that a helpful preprocessing step that you may wish to perform is to upsample your fMRI data if, e.g., the rate is not convenient with respect to your experiment.  For a variety of reasons (e.g. see Kay et al, Nature Methods, 2020), we believe this upsampling to be beneficial (slightly)
